service.rb

```diff
@@ -27,6 +27,26 @@ def initialize(port)
         end
         response_message << "</ul>\n"
         response_message << daily_data_form
+      when ["POST", "/add-daily-data"]
+        content_type = "text/html"
+
+        while not client.eof?
+          line = client.readline
+          if line.include?("Content-Length: ")
+            line.slice!("Content-Length: ")
+            content_length = line.to_i
+            break
+          end
+        end
```

Comment:
The effect of this section of code is to read everything up until the `Content-Length` header, extract its value and then bail out, probably with other headers still left to read. As a refactoring exercise it might be fun to replace it with a section of code which reads _all_ of the headers and e.g. leaves their values in a `headers` hash. That would make it a more self-contained (dare I say _modular_?) piece of functionality, plus it makes it more obvious later on that we’re reaching into the headers to extract the content length.